cmd/build/v2/checker/image_cache.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package checker
+
+import (
+	"context"
+	"sync"
+
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/registry"
+)
+
+type DigestResolver interface {
+	GetImageTagWithDigest(string) (string, error)
+}
+
+type Logger interface {
+	Infof(format string, args ...interface{})
+}
+
+type RegistryCacheProbe struct {
+	tagger      ImageTagger
+	namespace   string
+	registryURL string
+	imageCtrl   registry.ImageCtrl
+	registry    DigestResolver
+	logger      Logger
+
+	mu    sync.RWMutex
+	cache map[string]string
+}
+
+func NewImageChecker(tagger ImageTagger, namespace string, registryURL string, imageCtrl registry.ImageCtrl, registry DigestResolver, logger Logger) *RegistryCacheProbe {
```

Comment:
This is called `NewImageChecker` but it is returning `RegistryCacheProbe`, I'd update the name of the function to match what it really returns, specially having another struct that is called image checker